kombu/asynchronous/semaphore.py

```diff
@@ -28,13 +30,18 @@ class LaxBoundedSemaphore:
         HELLO 3
     """
 
-    def __init__(self, value):
+    def __init__(self, value: int) -> None:
         self.initial_value = self.value = value
         self._waiting = deque()
         self._add_waiter = self._waiting.append
         self._pop_waiter = self._waiting.popleft
 
-    def acquire(self, callback, *partial_args, **partial_kwargs):
+    def acquire(
+        self,
+        callback: Callable,
+        *partial_args: Any,
+        **partial_kwargs: Any
```

Comment:
```suggestion
        callback: Callable[P, None]
        *partial_args: P.args,
        **partial_kwargs: P.kwargs
```
<img width="812" alt="image" src="https://user-images.githubusercontent.com/7353520/162560074-35be31bc-d018-49db-ac0f-405e8875f906.png">

Reply:
I'd never heard of `ParamSpec` before - that's really cool!